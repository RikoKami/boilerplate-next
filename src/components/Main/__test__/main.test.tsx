import { render, screen } from "@testing-library/react"
import { Main } from ".."

describe("<Main />", () => {
	it("Should render the heading", () => {
		const { container } = render(<Main />)

		expect(
			// or screen.getByText(/react avançado/gi)
			screen.getByRole("heading", { name: /react avançado/gi })
		).toBeInTheDocument()
		expect(container.firstChild).toMatchSnapshot()
	})

	it("should render the colors correctly", () => {
		const { container } = render(<Main />)
		expect(container.firstChild).toHaveStyle({ "background-color": "#06092b" })
	})
})
