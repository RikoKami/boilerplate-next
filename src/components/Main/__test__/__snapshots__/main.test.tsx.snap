// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should render the heading 1`] = `
<main
  class="styles__Wrapper-dzx1ao-0 khDVDe"
>
  <img
    alt="Imagem de um átomo"
    class="styles__Logo-dzx1ao-1 jRpyFS"
    src="/img/icon-512.png"
  />
  <h1
    class="styles__Title-dzx1ao-2 lfuGAr"
  >
    React Avançado
  </h1>
  <h2
    class="styles__Description-dzx1ao-3 edoRUQ"
  >
    Typescript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="styles__Illustration-dzx1ao-4 cSlyKt"
    src="/img/hero-illustration.svg"
  />
</main>
`
